core/src/mast/mod.rs

```diff
@@ -94,9 +94,9 @@ impl MastForest {
         self.roots.iter().find(|&&root_id| self[root_id].digest() == digest).copied()
     }
 
-    /// Returns an iterator over the digest of the procedures in this MAST forest.
-    pub fn procedure_roots(&self) -> impl Iterator<Item = RpoDigest> + '_ {
-        self.roots.iter().map(|&root_id| self[root_id].digest())
+    /// Returns an iterator over the IDs of the procedures in this MAST forest.
+    pub fn procedure_roots(&self) -> &[MastNodeId] {
+        &self.roots
     }
```

Comment:
Same as last comment: I'd probably still keep the old method, just under `procedure_digests()` name and have `procedure_roots()` as you have them now.

Reply:
fixed